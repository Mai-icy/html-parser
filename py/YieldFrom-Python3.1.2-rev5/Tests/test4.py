#
#   Test delegation of next() call to subgenerator


def g1():
	print("Starting g1")
	yield "g1 ham"
	yield from g2()
	yield "g1 eggs"
	print("Finishing g1")

def g2():
	print("Starting g2")
	yield "g2 spam"
	yield "g2 more spam"
	print("Finishing g2")

for x in g1():
	print("Yielded", x)
