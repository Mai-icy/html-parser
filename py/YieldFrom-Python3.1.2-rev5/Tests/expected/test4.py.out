Starting g1
Yielded g1 ham
Starting g2
Yielded g2 spam
Yielded g2 more spam
Finishing g2
Yielded g1 eggs
Finishing g1
